lib/irc/ConnectionInstance.js

```diff
@@ -351,49 +362,12 @@ ConnectionInstance.create = function(server, opts, onCreatedCallback) {
                     `${opts.nick} is banned from ${server.domain}, ` +
                     `retrying in ${BANNED_TIME_MS}ms`
                 );
-
-                setTimeout(() => {
-                    retryFailed(
-                        new Error(`${opts.nick} was banned from ${server.domain}, retrying now`)
-                    );
-                }, BANNED_TIME_MS);
-                return;
-            }
-
-            if (server.getReconnectIntervalMs() > 0) {
-                // wait until scheduled
-                Scheduler.reschedule(server, retryTimeMs, retryConnection, opts.nick).then(
-                    returnClient,
-                    retryFailed
-                );
-            }
-            else {
-                let delay = BASE_RETRY_TIME_MS + retryTimeMs +
-                           Math.round((connAttempts * 1000) * Math.random());
-
-                log.info(`Scheduling disabled for ${opts.nick} on ${server.domain}, `+
-                         `retrying with delay ${delay}ms (attempts ${connAttempts})`);
-                retryConnection().then(
-                    returnClient,
-                    (e) => {
-                        setTimeout(
-                            () => {
-                                retryFailed(e);
-                            }
-                        , delay);
-                    }
-                );
+                yield Promise.delay(BANNED_TIME_MS);
             }
-
-        };
-
-        // Initially, pretend to have failed in order to start things off
-        retryFailed({});
+        }
     }
-    retryForever();
-
-    return d.promise;
-};
+    throw new Error("ConnectionInstance.create: failed to connect");
```

Comment:
Agreed. The intention is mainly for it to throw up in our faces if it ever did occur. I'll make the error here clearer.